packages/react/src/components/Authenticator/Router/index.tsx

```diff
@@ -46,7 +51,11 @@ export function Router({
 
   // `Authenticator` might not have `children` for non SPA use cases.
   if (['authenticated', 'signOut'].includes(route)) {
-    return children ? children({ signOut, user }) : null;
+    return children
+      ? typeof children === 'function'
+        ? children({ signOut, user }) // children is a React Node (e.g. <div>...</div>)
+        : children // children is a render prop
+      : null;
   }
```

Comment:
Nit-tish: nested ternaries can be difficult to reason with, can we early return if `children` is falsy and have a singular ternary below?
```suggestion
if (!children) {
  return null;
}
return typeof children === 'function'
  ? children({ signOut, user }) // children is a React Node (e.g. <div>...</div>)
  : children; // children is a render prop
```

Reply:
[64fcace](https://github.com/aws-amplify/amplify-ui/pull/1685/commits/64fcacef2d2a462fe76ababa26fdef6c99a587d5) for suggested comment; [e4f6985](https://github.com/aws-amplify/amplify-ui/pull/1685/commits/e4f69852a9ff9e33029317df275bae224ddf9cd6) for adding fragments around children

Reply:
Also realized I mixed up the comments, fixed in [eba4a95](https://github.com/aws-amplify/amplify-ui/pull/1685/commits/eba4a95f7ed35729ba2d107fe4ea7ac87631f9ef)